### ISSUE TITLE
After deleteing a user, RocketChat keep messages where user was mentioned

### ISSUE THREAD

**Issue by MichalNers** [RocketChat/Rocket.Chat#31950]
After deleting a user, there are still messages in the database with his user name where he was mentioned

**Comment by debdutdeb** [comment]
why should that change? If the message doesn't belong to the user, it shouldn't be deleted.

**Comment by MichalNers** [comment]
> why should that change? If the message doesn't belong to the user, it shouldn't be deleted.

yes, but it leaves information about the deleted user which is, for example inappropriate in the case of GDPR.
Orcan it be anonymised in some way?
![image](https://github.com/RocketChat/Rocket.Chat/assets/159016966/870ccdeb-2a89-4de7-b8c7-5453121e1b4b)

**Comment by debdutdeb** [comment]
type and _id both are essentially value-less here since the record being referenced doesn't exist. But, for username and name maybe. 

If you could rephrase the issue that way, I can bring it forward internally :)
